charts/kong/templates/_helpers.tpl

```diff
@@ -251,9 +251,11 @@ spec:
   {{- if .externalTrafficPolicy }}
   externalTrafficPolicy: {{ .externalTrafficPolicy }}
   {{- end }}
-  {{- if (and (eq .type "ClusterIP") .clusterIP )}}
+  {{- if .clusterIP }}
+  {{- if (or (not (eq .clusterIP "None")) (and (eq .type "ClusterIP") (eq .clusterIP "None"))) }}
```

Comment:
The last `(eq .clusterIP "None")` is not necessary because the `(or (not (eq .clusterIP "None"))...` will already imply it but it's OK cuz this way it's obvious ( or at least a bit more clear)

Reply:
To the extent that template truth tables are ever easy to visualize, anyway.